File: C++/Medium/Accepted/concatenation-of-consecutive-binary-numbers_6.cpp
class Solution {
public:
    int concatenatedBinary(int n) {
        
        long long int res = 0, mod = 1000000007;
        for (int i = 1; i <= n; i++) {
            int j = i;
            while (j) {
                res = (res << 1) % mod;
                j = j >> 1;
            }
            res = (res + i) % mod;
        }
        return res;
    }
};

File: Rust/Medium/WrongAnswer/concatenation-of-consecutive-binary-numbers_6.rs
impl Solution {
    pub fn concatenated_binary(n: i32) -> i32 {
        
        let mut res = 0;
        for i in 1..=n {
            res = (res * (i as i64).leading_zeros() as i64 + i as i64) % 1000000007;
        }
        res as i32
    }
}